osu.Server.BeatmapSubmission/Services/BeatmapPackageParser.cs

```diff
@@ -18,18 +18,36 @@ namespace osu.Server.BeatmapSubmission.Services
 {
     public static class BeatmapPackageParser
     {
+        public static readonly HashSet<string> INVALID_EXTENSIONS = new HashSet<string>(StringComparer.OrdinalIgnoreCase)
+        {
+            ".exe", ".cmd", ".bat", ".sh", ".scr", ".doc", ".docx", ".docm", ".hta", ".htm", ".html", ".js", ".jar", ".vbs", ".vb", ".pdf", ".sfx", ".dll", ".py",
```

Comment:
a few more we could probably consider adding: `.cpl`, `.pif`, `.com`, `.wsh`, `.ps1`, `.msc`, `.iso`, `.zip`... and every other archive forma... you know, maybe this should be a whitelist instead of a blacklist...?

Reply:
`AudioCheckUtils.AUDIO_EXTENSIONS`, `OsuGameBase.VIDEO_EXTENSIONS`

might make sense to centralise these (and combine the multiple specifications of `IMAGE_EXTENSIONS` without constants in the `osu.Game` codebase).

Reply:
> you know, maybe this should be a whitelist instead of a blacklist...?

Yeah I considered it. My biggest hangup was that I wasn't confident that I wasn't gonna spill the baby with the bathwater, as in exclude formats we don't have listed that the game does support. For one the game checks for image extensions are very limited, it's just JPG and PNG everywhere. That's probably gonna cover 90% of use cases but I'm not sure it'll cover 100%.

If you think that's fine then I'm ok with doing that, but I am wary of future complaints of "why is this BSS so constraining I can't use some valid image formats" or something.

Reply:
Applied in https://github.com/ppy/osu-server-beatmap-submission/pull/5/commits/0fe7ed7abe3b23b04d80639bcec96b4d41be165b